frontend/src/TableFilter.tsx

```diff
@@ -83,3 +118,11 @@ export default function TableFilter<O extends string | MenuOption>({
         </th>
     );
 }
+
+function isOptionEqual(a?: Option, b?: Option) {
```

Comment:
Why are the arguments to this equality checker optional? When might I want to compare something to nothing, or nothing to nothing?

Reply:
The `allOption` and `emptyOption` passed into `TableFilter` are optional; making the equality checker allow undefined options saves some repeated checking of whether or not `allOption` and `emptyOption` exist. (But actually in looking at it, only the second arg actually needs to be optional to get this benefit, so I'll remove the `?` from the first one!)

Reply:
P.S. ended up making the second arg required too as part of adjusting the code for your other comment - it ended up a lot clearer I think 🎉